source/common/tracing/BUILD

```diff
@@ -12,12 +12,16 @@ envoy_cc_library(
     name = "http_tracer_lib",
     srcs = [
         "http_tracer_impl.cc",
-        "lightstep_tracer_impl.cc",
-    ],
+    ] + select({
+        "//bazel:disable_lightstep_tracing": [],
+        "//conditions:default": ["lightstep_tracer_impl.cc"],
+    }),
     hdrs = [
         "http_tracer_impl.h",
-        "lightstep_tracer_impl.h",
-    ],
+    ] + select({
+        "//bazel:disable_lightstep_tracing": [],
+        "//conditions:default": ["lightstep_tracer_impl.h"],
+    }),
     external_deps = ["lightstep"],
```

Comment:
Is it possible to separate `lightstep_tracer_impl.{cc,h}` as a separate `cc_library` and use do `select` in `deps`?
It will help our use case in istio proxy so we don't need build lightstep at all, cc #840